.github/workflows/static_analysis.yaml

```diff
@@ -0,0 +1,49 @@
+name: static_analysis
+
+on:
+  push:
+    branches: ['**']
+  pull_request:
+    branches: ['**']
+
+env:
+  # Expected error count should decrease as we handle the errors. Update accordingly.
+  CPPCHECK_EXPECTED_ERROR_COUNT: 53
+
+jobs:
+  cppcheck:
+    name: cppcheck
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Install 
+        shell: bash
+        run: |
+          sudo add-apt-repository --yes ppa:ubuntu-toolchain-r/test;
+          sudo apt-get update;
+          sudo apt-get install -qq cppcheck;
+      
+      - name: Run
+        shell: bash
+        run: | 
+          cppcheck . --enable=performance,portability,missingInclude --std=c++11 -ithird-party/easyloggingpp -ithird-party/glad -ithird-party/sqlite -ithird-party/imgui -ithird-party/glfw -ithird-party/hidapi -ithird-party/json.hpp --xml &> cppcheck_run.log
```

Comment:
We can remove `performance,portability` right?
I think when I ran it locally I needed to add some include explicitly, don't we need it here?

Reply:
`-I src/ -I include/ -I common/  `